tests/Pipeline/PipelineTest.php

```diff
@@ -249,6 +250,34 @@ public function testPipelineThenReturnMethodRunsPipelineThenReturnsPassable()
 
         unset($_SERVER['__test.pipe.one']);
     }
+
+    public function testItThrowsInvalidArgumentExceptionWhenPipeIsOfSillyTypes()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('A pipe should of type object, string or a callable. integer give');
```

Comment:
I suggest. "A pipe must be an object, a string or a callable. %s given."

Reply:
You mean check it by regex ?

Reply:
I mean changing the whole message. Your current message grammar is not correct.

Reply:
Yeah, nice point. applied.